main/solution/backend/src/lambdas/data-source-reachability/handler.js

```diff
@@ -25,9 +27,32 @@ const handler = async () => {
   await registerServices(container, pluginRegistry);
   await container.initServices();
   const dataSourceReachabilityService = await container.find('dataSourceReachabilityService');
+  const studyService = await container.find('studyService');
+  const dataSourceAccountService = await container.find('dataSourceAccountService');
   const userContext = getSystemRequestContext();
+
+  // Check reachability of all Data Source Accounts ONLY in unreachable states
+  // This kicks off reachability check workflows for all its associated DS studies only if the DS Account status changes
   await dataSourceReachabilityService.attemptReach(userContext, { id: '*', status: 'pending' });
   await dataSourceReachabilityService.attemptReach(userContext, { id: '*', status: 'error' });
+
+  // Check reachability of all Data Source Studies ONLY in unreachable states
```

Comment:
If an unreachable study is becoming reachable, the opposite might also be true, right? I guess the status would eventually refresh?

Reply:
When a reachable study becomes unreachable, it's status in the DB won't change until the next force check (occurs once every day via `data-source-reachability-daily` lambda).

Reply:
I think one good to have feature would be to add some metadata in UI which shows when was the last check performed regardless of its status. Seems misleading to call something reachable if it isn't reachable and refresh it after a day. Probably something to add in some follow up ticket?

Reply:
This already exists and the user can view this when expanding the study card.